crates/hotshot-state-prover/src/circuit/types.rs

```diff
@@ -0,0 +1,40 @@
+//! Types for verification circuit
+
+use ark_ff::PrimeField;
+pub use jf_primitives::circuit::signature::schnorr::{SignatureVar, VerKeyVar};
+use jf_relation::Variable;
+use serde::{Deserialize, Serialize};
+
+/// Variable for stake table entry
+#[derive(Clone, Debug)]
+pub struct StakeTableEntryVar {
+    pub bls_ver_key: (Variable, Variable),
+    pub schnorr_ver_key: VerKeyVar,
+    pub stake_amount: Variable,
+}
+
+/// HotShot state Variable
+#[derive(Clone, Debug)]
+pub struct HotShotStateVar {
```

Comment:
Maybe add a From() method converting between `HotShotStateVar` and `HotshotState`.

Reply:
Such `From()` also needs to take a circuit as input. Doesn't seems possible.